src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -53,7 +53,7 @@ function ($deferred, $namespace, &$expiredIds) {
 
                 foreach ($deferred as $key => $item) {
                     if (null === $item->expiry) {
-                        $byLifetime[0][$namespace.$key] = $item->value;
+                        $byLifetime[0 < $item->defaultLifetime ? $item->defaultLifetime : 0][$namespace.$key] = $item->value;
```

Comment:
Maybe `max(0, $item->defaultLifetime)` would be more readable?  Maybe it's not a good idea because it implies a function call?